source/unified-test-format/unified-test-format.rst

```diff
@@ -2492,7 +2560,224 @@ An example of this operation follows::
         client: *client0
         connections: 1
 
+runOnThread
```

Comment:
I didn't previously consider this, but the introduction of `runOnThread` means the EntityMap may now need to be a thread-safe container. That may warrant special mention in the earlier [Entity Map](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#entity-map) section, as it'd likely be a non-trivial change for some drivers.

This isn't just about thread-safe access. I'm also thinking about an operation potentially writing something to the entity map from a thread.

Reply:
Yep good point. Making the entity map thread safe was probably the most involved change I needed to make in the Rust runner (especially given the Rust-specific things the compiler enforces). Added a note about it.